test/games/strategy/engine/data/CardDeckTest.java

```diff
@@ -0,0 +1,64 @@
+package games.strategy.engine.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import junit.framework.TestCase;
+
+public class CardDeckTest extends TestCase {
+
+  private CardDeck cardDeck;
+  final int deckSizeAtStart = 10;
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+
+    final List<Card> cards = new ArrayList<Card>(deckSizeAtStart);
+    for (int currentCard = 1; currentCard <= deckSizeAtStart; ++currentCard) {
+      cards.add(new Card("Card " + currentCard, PlayerID.NULL_PLAYERID, null));
+    }
+    cardDeck = new CardDeck(cards);
+  }
+
+  @Test
+  public void testCardDeck() {
+    assertNotNull(cardDeck);
+    assertEquals(deckSizeAtStart, cardDeck.getDeckSize());
+  }
+
+  @Test
+  public void testGetStacksForAllCards() {
+    final int stacksCounter = 4;
+    final List<List<Card>> stacks = cardDeck.getStacksForAllCards(stacksCounter);
+    // Deck should be empty now
+    assertEquals(0, cardDeck.getDeckSize());

```

Comment:
recommend using hamcrest. This would be:
`assertThat(cardDeck.getDeckSize(), is(0))`